### PR TITLE
Add controllers and corresponding views

### PR DESCRIPTION
**Pull request by ReemMohamedAbdelfatah**
Hello 👋 In this  PR the following is done:

- [x] Created required routes in Config/routes.rb
- [x] Created Corresponding Controllers
- [x] Created required views

### CODE REVIEW

File: spec/helpers/posts_helper_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# describe PostsHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe PostsHelper, type: :helper do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (omar-muhamad) [inline spec/helpers/posts_helper_spec.rb:15]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Setup_Controllers_project_v1_1.md#:~:text=Important%3A-,do%20NOT%20use%20rails%20generate%20scaffold%20command.,-The%20point%20of), You specifically asked not to use `rails generate scaffold` command and you used it anyway.
Please remove excess files created by the command like:
   - All created files in `./app/helpers`.
   - All created files in `./spec`.
   
   
image

image

File: config/routes.rb [hunk 2]
```diff
@@ -1,4 +1,11 @@
 Rails.application.routes.draw do
+  resources :users
+  resources :posts
+  root to: 'users#index'
+  get 'posts/index'
+  get 'posts/show'
+  get 'users/index'
+  get 'users/show'
```
Comment (omar-muhamad) [inline config/routes.rb]: - [ ] Posts routes are not properly configured as requested in the project requirements and also you created some routes more than once. Please configure your routes correctly to match the given `rails routes` command output in the project requirement.

image
```suggestion
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
end
  
root to: 'users#index'
```

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @ReemMohamedAbdelfatah,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Reem-lab — APPROVED** [review]
# Hi @ReemMohamedAbdelfatah,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  Descriptive PR⭐
- [ ]  No Linters error 🤩
- [ ] Good job working on the controllers💪🏻
- [ ] Routes set up well🔥🔥



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 2 above)
Comment (ReemMohamedAbdelfatah) [reply config/routes.rb]: I used "rails generate controller Users index show" is it considered scaffold?

File: spec/helpers/posts_helper_spec.rb (hunk 1 above)
Comment (ReemMohamedAbdelfatah) [reply spec/helpers/posts_helper_spec.rb:15]: I used "rails generate controller Users index show" is it considered scaffold?

**Comment by ReemMohamedAbdelfatah** [comment]
Thank you

**Pull request merged**